src/sentry/api/endpoints/organization_teams.py

```diff
@@ -169,29 +187,41 @@ def post(self, request: Request, organization, **kwargs) -> Response:
                     },
                     status=409,
                 )
-            else:
-                team_created.send_robust(
-                    organization=organization, user=request.user, team=team, sender=self.__class__
-                )
             if self.should_add_creator_to_team(request):
                 try:
                     member = OrganizationMember.objects.get(
                         user=request.user, organization=organization
                     )
+                    OrganizationMemberTeam.objects.create(
+                        team=team,
+                        organizationmember=member,
+                        role="admin" if set_team_admin else None,
+                    )
                 except OrganizationMember.DoesNotExist:
-                    pass
-                else:
-                    OrganizationMemberTeam.objects.create(team=team, organizationmember=member)
-
-            self.create_audit_entry(
-                request=request,
-                organization=organization,
-                target_object=team.id,
-                event=audit_log.get_event_id("TEAM_ADD"),
-                data=team.get_audit_log_data(),
-            )
-            return Response(
-                serialize(team, request.user, self.team_serializer_for_post()),
-                status=201,
-            )
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+                    # Only rollback team creation and return 400 if trying to set user as
+                    # Team Admin but user is not a member of the organization
+                    if set_team_admin:
+                        transaction.set_rollback(True)
```

Comment:
If you move the exception blocks out of the atomic block, then you don't need to do this rollback explicitly. Django ORM should do it for you.